arraySort.js

```diff
@@ -1,31 +1,34 @@
-var evenNumbers = new Array();
-var oddNumbers = new Array();
-var characters = new Array();
+function arraySort(myArray) {
+    var evenNumbers = new Array();
+    var oddNumbers = new Array();
+    var characters = new Array();
 
-let numbers = {
-    evens: evenNumbers,
-    odds: oddNumbers,
-    chars: characters,
-}
+    let numbers = {
+        evens: evenNumbers,
+        odds: oddNumbers,
+        chars: characters,
+    }
 
-function arraySort(myArray) {
-    for (var i = 0; i < myArray.length; i++) {
-        if (isNaN(myArray[i])) {
-            characters.push(myArray[i]);
-        }
-        if (myArray[i] % 2 == 0) {
-            evenNumbers.push(myArray[i]);
-        }
-        if (myArray[i] % 2 == 1) {
-            oddNumbers.push(myArray[i]);
+    if (!Array.isArray(myArray)) {
+        throw new Error('invalid input')
+    }
+    else {
```

Comment:
By the way if an error is thrown, The function stops executing so, the `else` block here is redundant and you can ignore it